Source/Fluids/WarpXFluidContainer.cpp

```diff
@@ -0,0 +1,194 @@
+                                 amrex::MultiFab *rho, amrex::MultiFab *crho,
+                                 const amrex::MultiFab *cEx, const amrex::MultiFab *cEy, const amrex::MultiFab *cEz,
+                                 const amrex::MultiFab *cBx, const amrex::MultiFab *cBy, const amrex::MultiFab *cBz,
+                                 amrex::Real t, amrex::Real dt, DtType a_dt_type, bool skip_deposition)
+{
+    // Gather E&B fields to each node
+    // TODO
+
+    // Update N and NU at each node
+    // TODO
+
+    // Deposit J to the simulation mesh
+    if (!skip_deposition)
+    {
+        DepositCurrent(lev, jx, jy, jz);
+    }
+}
+
+void WarpXFluidContainer::DepositCurrent(int lev,
+                                         amrex::MultiFab &jx, amrex::MultiFab &jy, amrex::MultiFab &jz)
+{
+
+    const amrex::ParticleReal inv_clight_sq = 1.0_prt / PhysConst::c / PhysConst::c;
+    charge = plasma_injector->getCharge();
+
+// Loop over grid
+#ifdef AMREX_USE_OMP
+#pragma omp parallel if (amrex::Gpu::notInLaunchRegion())
+#endif
+    for (MFIter mfi(*N[lev], TilingIfNotGPU()); mfi.isValid(); ++mfi)
+    {
+
+        amrex::Box const &tile_box = mfi.tilebox(N[lev]->ixType().toIntVect());
+        amrex::Array4<Real> const &N_arr = N[lev]->array(mfi);
+        amrex::Array4<Real> const &NUx_arr = NU[lev][0]->array(mfi);
+        amrex::Array4<Real> const &NUy_arr = NU[lev][1]->array(mfi);
+        amrex::Array4<Real> const &NUz_arr = NU[lev][2]->array(mfi);
+
+        amrex::Array4<amrex::Real> jx_arr = jx.array(mfi);
+        amrex::Array4<amrex::Real> jy_arr = jy.array(mfi);
+        amrex::Array4<amrex::Real> jz_arr = jz.array(mfi);
+
+        // Loop over cells (ParallelFor)
+        amrex::ParallelFor(tile_box, [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept
+                           {
+                               // Interpolate N/NU from nodes to the simulation mesh (typically Yee mesh)
+                               amrex::GpuArray<int, 3U> sf = {AMREX_D_DECL(1, 1, 1)};
+                               auto N_CC = ablastr::coarsen::sample::Interp(N_arr, sf, sf, sf, i, j, k, 1);
+                               auto NUx_CC = ablastr::coarsen::sample::Interp(NUx_arr, sf, sf, sf, i, j, k, 1);
+                               auto NUy_CC = ablastr::coarsen::sample::Interp(NUy_arr, sf, sf, sf, i, j, k, 1);
+                               auto NUz_CC = ablastr::coarsen::sample::Interp(NUz_arr, sf, sf, sf, i, j, k, 1);
+
+                               // Calculate J from fluid and add it to jx/jy/jz
+                               auto gamma = std::sqrt(N_CC * N_CC + (NUx_CC * NUx_CC + NUy_CC * NUy_CC + NUz_CC * NUz_CC)*inv_clight_sq) / N_CC;
+                               jx_arr(i, j, k) += charge * (NUx_CC / gamma);
+                               jy_arr(i, j, k) += charge * (NUy_CC / gamma);
+                               jz_arr(i, j, k) += charge * (NUz_CC / gamma); });
```

Comment:
In order to be consistent with the above suggestion:
```suggestion
                               jx_arr(i, j, k) += q * (NUx_CC / gamma);
                               jy_arr(i, j, k) += q * (NUy_CC / gamma);
                               jz_arr(i, j, k) += q * (NUz_CC / gamma); });
```

Reply:
Changed.